src/test/java/org/folio/rest/impl/NotificationsResourceImplTest.java

```diff
@@ -0,0 +1,795 @@
+package org.folio.rest.impl;
+
+import static io.vertx.core.Future.failedFuture;
+import static io.vertx.core.Future.succeededFuture;
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.core.Response;
+
+import org.awaitility.Awaitility;
+import org.folio.client.OkapiModulesClient;
+import org.folio.rest.RestVerticle;
+import org.folio.rest.jaxrs.model.EventEntity;
+import org.folio.rest.jaxrs.model.Notification;
+import org.folio.rest.jaxrs.model.ResultInfo;
+import org.folio.rest.persist.PostgresClient;
+import org.folio.rest.persist.cql.CQLWrapper;
+import org.folio.rest.persist.interfaces.Results;
+import org.folio.rest.tools.client.interfaces.HttpClientInterface;
+import org.folio.rest.tools.utils.ObjectMapperTool;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Handler;
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import io.vertx.ext.sql.UpdateResult;
+
+@RunWith(MockitoJUnitRunner.class)
+public class NotificationsResourceImplTest {
+  private final Logger logger = LoggerFactory.getLogger(NotificationsResourceImplTest.class);
+
+  private static final String USER_ID = UUID.randomUUID().toString();
+  private static final String USERNAME = "username";
+  private static final String LANG = "en";
+
+  @Spy
+  private NotificationsResourceImpl notificationsResource;
+
+  @Mock
+  private HttpClientInterface httpClient;
+
+  @Mock
+  private PostgresClient postgresClient;
+
+  @Mock
+  private OkapiModulesClient okapiModulesClient;
+
+  @Mock
+  private Notification notification;
+
+  private boolean handlerIsCalled;
+  private Map<String, String> okapiHeaders = new HashMap<>();
+  private int status;
+
+  @Before
+  public void setUp() {
+    handlerIsCalled = false;
+
+    doReturn(httpClient).when(notificationsResource).getHttpClient(any(), any());
+    doReturn(postgresClient).when(notificationsResource).getPostgresClient(any(), any());
+    doReturn(okapiModulesClient).when(notificationsResource).getOkapiModulesClient(any(), any());
```

Comment:
Mockito documentation says:
> Use `doReturn()` in those rare occasions when you cannot use `Mockito#when()`. Beware that `Mockito#when()` is always recommended for stubbing because it is argument type-safe and more readable (especially when stubbing consecutive calls).

Reply:
Sounds good to me, it was just an FYI.